test/sys/test_timerfd.rs

```diff
@@ -0,0 +1,49 @@
+use nix::errno::Errno;
+use nix::sys::time::{TimeSpec, TimeValLike};
+use nix::sys::timerfd::{ClockId, Expiration, TimerFd, TimerFlags, TimerSetTimeFlags};
+use nix::unistd::read;
+use nix::Error;
+use std::os::unix::io::AsRawFd;
+use std::time::Instant;
+
+#[test]
+pub fn test_timerfd_oneshot() {
+    let mut buffer = [0u8; 64];
+    let fd = TimerFd::new(ClockId::Monotonic, TimerFlags::empty()).unwrap();
+
+    let before = Instant::now();
+
+    fd.settime(
+        TimerSetTimeFlags::empty(),
+        Expiration::OneShot(TimeSpec::seconds(1)),
+    )
+    .unwrap();
+
+    read(fd.as_raw_fd(), &mut buffer).unwrap();
+
+    let millis = before.elapsed().as_millis();
+    assert!(millis < 1100 && millis > 900);
+}
+
+#[test]
+pub fn test_timerfd_interval() {
+    let mut buffer = [0u8; 64];
+    let fd = TimerFd::new(ClockId::Monotonic, TimerFlags::empty()).unwrap();
+
+    let before = Instant::now();
+    fd.settime(
+        TimerSetTimeFlags::empty(),
+        Expiration::IntervalDelayed(TimeSpec::seconds(1), TimeSpec::seconds(2)),
+    )
+    .unwrap();
+
+    read(fd.as_raw_fd(), &mut buffer).unwrap();
+
+    let start_delay = before.elapsed().as_millis();
+    assert!(start_delay < 1100 && start_delay > 900);
```

Comment:
Ditto as above.  You can't assert an upper bound for the delay.

Reply:
Same as above.